cmd/statusd/main.go

```diff
@@ -158,6 +166,21 @@ func startCollectingStats(interruptCh <-chan struct{}, nodeManager common.NodeMa
 	}
 }
 
+func syncAndStopNode(nodeManager common.NodeManager) {
+	log.Println("Node will synchronize and exit")
+	err := nodeManager.Sync((time.Duration)(*syncAndExit) * time.Minute)
+	if err != nil {
+		log.Fatalf("Failed while waiting for sync: %v", err)
```

Comment:
Yeah, as @adambabik suggested, use `errors.Wrap` here and in line 179 and return error from the function.